extensions/pkg/terraformer/config.go

```diff
@@ -245,17 +246,29 @@ func (t *terraformer) CleanupConfiguration(ctx context.Context) error {
 	t.logger.Info("Cleaning up all terraformer configuration")
 
 	t.logger.V(1).Info("Deleting Terraform variables Secret", "name", t.variablesName)
-	if err := t.client.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}); err != nil && !apierrors.IsNotFound(err) {
+	varObject := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}
+	if err := extensionscontroller.DeleteAllFinalizers(ctx, t.client, varObject); err != nil {
```

Comment:
Shouldn't we remove only the `gardener.cloud/terraformer` finalizer?

Reply:
Good catch.